test/function_building_error_messages.jl

```diff
@@ -309,6 +309,7 @@ DAEFunction(dfoop, vjp = dvjp)
 dvjp(res, du, u, v, p, gamma, t) = [1.0]
 DAEFunction(dfiip, vjp = dvjp)
 DAEFunction(dfoop, vjp = dvjp)
+DAEFunction{true, SciMLBase.NoSpecialize}(dfiip, observed = 1)
```

Comment:
that's not a valid observed function?

Reply:
all this is testing is that the observed gets put in the right place. Before this would error because it would try to put the the `observed` into the spot for `syms` and break.

Reply:
This should check the function form and throw an error... in the future.